cfc-nanny/layers/rcon_dependencies/rcon_status_parser.py

```diff
@@ -0,0 +1,86 @@
+# https://github.com/CFC-Servers/rcon_status_parser
+import json
+import re
+
+class StatusParser():
+    def __init__(self):
+        self.patterns = {
+            "server": {
+                "hostname"        : re.compile('hostname(?: *):(?: *)(.*)'),
+                "version"         : re.compile('version(?: *):(?: *)(.*)'),
+                "ip"              : re.compile('udp/ip(?: *):(?: *)(.*):\d'),
+                "port"            : re.compile('udp/ip(?: *):(?: *).*:(\d+) '),
+                "map"             : re.compile('map(?: *):(?: *)(\w*)'),
+                "player_count"    : re.compile('players(?: *):(?: *)(\d+)'),
+                "max_player_count": re.compile('players(?: *):(?: *).*\((\d+)')
+            },
+            "players": {
+                "id"              : re.compile('(?: *)(\d+)(?: *)\"'),
+                "name"            : re.compile('\d+ \"(.*)\"'),
+                "steam_id"        : re.compile('(STEAM_\d:\d:\d+)(?: *)'),
+                "time_connected"  : re.compile(' ((?:\d{1,2}:)?\d{2}:\d{2}) '),
+                "ping/loss"       : re.compile(' (\d+) '),
+                "state"           : re.compile('(active|spawning)'),
+                "ip"              : re.compile('(?: *)((?:\d{1,3}\.?){4}):')
+            }
+        }
+
+    def parse(self, rcon):
+
+        return {
+            "hostname"         : self.get_hostname(rcon),
+            "version"          : self.get_version(rcon),
+            "ip"               : self.get_ip(rcon),
+            "port"             : self.get_port(rcon),
+            "map"              : self.get_map(rcon),
+            "player_count"     : self.get_player_count(rcon),
+            "max_player_count" : self.get_max_player_count(rcon),
+            "players"          : self.get_players(rcon)
+        }
+
+    def get_hostname(self, rcon):
+        return re.findall(self.patterns["server"]["hostname"], rcon)[0]
```

Comment:
I think you can also do
```suggestion
        return self.patterns["server"]["hostname"].findall(rcon)[0]
```